synapse/tests/test_lib_cell.py

```diff
@@ -122,3 +126,306 @@ class CellTest(s_t_utils.SynTest):
+                headers = {'origin': origin}
+                async with sess.get(f'https://localhost:{port}/api/v1/auth/users', headers=headers) as resp:
+                    retn = await resp.json()
+                    self.eq(origin, resp.headers.get('Access-Control-Allow-Origin'))
+                    self.eq('ok', retn.get('status'))
+
+                # use the authenticated session to do stuff...
+                async with sess.get(f'https://localhost:{port}/api/v1/auth/users') as resp:
+                    retn = await resp.json()
+                    self.eq('ok', retn.get('status'))
+
+                info = {'rules': ()}
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/user/{visiiden}', json=info) as resp:
+                    retn = await resp.json()
+                    self.eq('ok', retn.get('status'))
+
+                info = {'rules': ()}
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/role/{analystiden}', json=info) as resp:
+                    retn = await resp.json()
+                    self.eq('ok', retn.get('status'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/adduser', json={}) as resp:
+                    item = await resp.json()
+                    self.eq('MissingField', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/addrole', json={}) as resp:
+                    item = await resp.json()
+                    self.eq('MissingField', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/adduser', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/addrole', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/grant', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/revoke', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/user/{visiiden}', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.post(f'https://localhost:{port}/api/v1/auth/role/{analystiden}', data=b'asdf') as resp:
+                    item = await resp.json()
+                    self.eq('BadJson', item.get('code'))
+
+                async with sess.get(f'https://localhost:{port}/api/v1/storm', data=b'asdf') as resp:
```

Comment:
testing /v1/storm and /v1/storm/nodes here feels incorrect. If/when we need to extend tests for that API, we'll have to carve them out separately into their own.

Reply:
moving all httpapi tests into synapse/tests/test_lib_httpapi.py